src/signals/settings/users/containers/Overview/__tests__/Overview.test.js

```diff
@@ -440,4 +422,176 @@ describe('signals/settings/users/containers/Overview', () => {
 
     expect(dispatch).toHaveBeenCalledTimes(1);
   });
+
+  it('should render a role filter', async () => {
+    const { getByTestId } = render(usersOverviewWithAppContext());
+
+    await wait();
+
+    expect(getByTestId('roleSelect')).toBeInTheDocument();
+  });
+
+  it('should select the right option when role filter changes', async () => {
+    jest
+      .spyOn(appSelectors, 'makeSelectUserCan')
+      .mockImplementation(() => () => true);
+
+    jest
+      .spyOn(rolesSelectors, 'inputRolesSelector')
+      .mockImplementation(() => inputRolesSelectorJSON);
+
+    const { getByTestId } = render(usersOverviewWithAppContext());
+
+    await wait();
+
+    const filterByRoleSelect = getByTestId('roleSelect');
+
+    // check if the default value has been set
+    expect(filterByRoleSelect.value).toBe('*');
+
+    const filterValue = 'Regievoerder';
+
+    act(() => {
+      fireEvent.change(filterByRoleSelect, { target: { value: filterValue } });
+    });
+
+    expect(dispatch).toHaveBeenCalledTimes(1);
```

Comment:
Can you assert that `dispatch` has not been called before the event was fired? For good measure.